### PR TITLE
fix(inputs.phpfpm): Use logger without causing panic

### LINKED ISSUE #14485: phpfpm input plugin panics in fcgi mode with json format
### Relevant telegraf.conf

```toml
[[inputs.phpfpm]]
  urls = ["fcgi://php-fpm:9001/status?full&json"]
  format = "json"
```

### Logs from Telegraf

```text
2023-12-20T12:29:49Z I! Loading config: /etc/telegraf/telegraf.conf
2023-12-20T12:29:49Z I! Starting Telegraf 1.29.1 brought to you by InfluxData the makers of InfluxDB
2023-12-20T12:29:49Z I! Available plugins: 241 inputs, 9 aggregators, 30 processors, 24 parsers, 60 outputs, 6 secret-stores
2023-12-20T12:29:49Z I! Loaded inputs: phpfpm
2023-12-20T12:29:49Z I! Loaded aggregators: 
2023-12-20T12:29:49Z I! Loaded processors: 
2023-12-20T12:29:49Z I! Loaded secretstores: 
2023-12-20T12:29:49Z I! Loaded outputs: prometheus_client
2023-12-20T12:29:49Z I! Tags enabled: host=e42eae5ec81c
2023-12-20T12:29:49Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"e42eae5ec81c", Flush Interval:10s
2023-12-20T12:29:49Z D! [agent] Initializing plugins
2023-12-20T12:29:49Z D! [agent] Connecting outputs
2023-12-20T12:29:49Z D! [agent] Attempting connection to [outputs.prometheus_client]
2023-12-20T12:29:49Z I! [outputs.prometheus_client] Listening on http://[::]:9273/metrics
2023-12-20T12:29:49Z D! [agent] Successfully connected to outputs.prometheus_client
2023-12-20T12:29:49Z D! [agent] Starting service inputs
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x3392cd1]
```

### System info

Telegraf 1.29.1, Ubuntu 22.04, Docker 24.0.7, Docker Compose 2.21.0

### Docker

```yaml
# docker-compose.yml
services:
  php-fpm:
    image: docker.io/library/php:8.3.0-fpm-alpine3.19
    volumes:
      - ./zz-php-fpm.conf:/usr/local/etc/php-fpm.d/zz-php-fpm.conf:ro
  telegraf83:
    image: docker.io/library/telegraf:1.29.1-alpine
    entrypoint:
      - /entrypoint.sh
    command:
      - telegraf
      - --debug
    volumes:
      - ./telegraf.conf:/etc/telegraf/telegraf.conf:ro
```

```ini
# zz-php-fpm.conf
[www]
pm.status_listen = 9001
pm.status_path = /status
```

```toml
[[inputs.phpfpm]]
  urls = ["fcgi://php-fpm:9001/status?full&json"]
  format = "json"
```

### Steps to reproduce

1. `docker compose up --detach`
2. `docker compose logs --follow telegraf`

### Expected behavior

No panic.

### Actual behavior

Panic.

### Additional info

When I add a Nginx FastCGI proxy between Telegraf and PHP-FPM and configure Telegraf to use HTTP to get the metrics there is no panic (see log output below).

I guess one issue is that the [gatherFcgi](https://github.com/influxdata/telegraf/blob/9a469977dd46c73a91dfaf73dd306d6a7dd1ec7a/plugins/inputs/phpfpm/phpfpm.go#L193) method is not setting the QUERY_STRING (`full&json` in my case). It might also be necessary to remove the query (`?full&json`) from the `statusPath` when setting `SCRIPT_NAME` and `SCRIPT_FILENAME`.

```
2023-12-20T12:29:50Z I! Loading config: /etc/telegraf/telegraf.conf
2023-12-20T12:29:50Z I! Starting Telegraf 1.29.1 brought to you by InfluxData the makers of InfluxDB
2023-12-20T12:29:50Z I! Available plugins: 241 inputs, 9 aggregators, 30 processors, 24 parsers, 60 outputs, 6 secret-stores
2023-12-20T12:29:50Z I! Loaded inputs: phpfpm
2023-12-20T12:29:50Z I! Loaded aggregators: 
2023-12-20T12:29:50Z I! Loaded processors: 
2023-12-20T12:29:50Z I! Loaded secretstores: 
2023-12-20T12:29:50Z I! Loaded outputs: prometheus_client
2023-12-20T12:29:50Z I! Tags enabled: host=c72f20970a05
2023-12-20T12:29:50Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"c72f20970a05", Flush Interval:10s
2023-12-20T12:29:50Z I! [outputs.prometheus_client] Listening on http://[::]:9273/metrics
``

### PR DESCRIPTION
**Pull request by elchenberg**
## Summary

Use logger without causing panic.

## Checklist

- [x] No AI generated code was used in this PR

## Related issues

Resolves #14485

### CODE REVIEW

**Comment by elchenberg** [comment]
!signed-cla

File: plugins/inputs/phpfpm/phpfpm_test.go [hunk 1]
```diff
@@ -358,3 +359,14 @@ max active processes: 1
...
+func TestPhpFpmLogWithoutPanic(t *testing.T) {
+	p := &phpfpm{}
+	// AddInput sets the Logger
+	if err := shim.New().AddInput(p); err != nil {
+		t.Error(err)
+		return
+	}
+
+	p.Log.Debug("test")
+}
```
Comment (powersj) [inline plugins/inputs/phpfpm/phpfpm_test.go:372]: I'd prefer we drop this test, as this is not something we have in every other PR. And instead put up a test case that fails to decode the JSON, which would hit the log error.

**Review by powersj — COMMENTED** [review]
Thanks for the PR - one comment inline

File: plugins/inputs/phpfpm/phpfpm_test.go (hunk 1 above)
Comment (elchenberg) [reply plugins/inputs/phpfpm/phpfpm_test.go:391]: I changed the test as suggested.

**Review by powersj — APPROVED** [review]
Thank you!

**Pull request merged**